Make runnable as GitHub action

sbrudz opened the pull request:
This PR switches the tool to run as a GitHub action instead of as a command line tool, allowing the report to be generated on a regular basis.

sbrudz committed:
ci: configure CI build using github actions

sbrudz committed:
ci: run build, formatting, and linting as part of CI build

sbrudz committed:
feat: provide report generation as a github action

sbrudz committed:
refactor: retrieve heroku_api_token from action inputs instead of environment variable

sbrudz committed:
refactor: retrieve git_project_directory from action inputs instead of environment variable

sbrudz committed:
fix: switch to node12 for action runner

because node14 is not yet supported

sbrudz committed:
build: commit contents of dist directory

so that this can be used as a public action

sbrudz committed:
fix: fully downgrade to node 12

because "fs/promises" package only exists in node 14 and we can't use that in github actions yet

its-fern commented on .nvmrc:
```diff
@@ -1 +1 @@
-14.4.0
+12.14
```
This version conflicts with the version in the github [workflow file](https://github.com/sbrudz/accelerate-metrics/pull/4/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R12) - is that intended?

sbrudz replied:
That was an oversight.  Good catch!

sbrudz replied:
Fixed

its-fern commented on .gitignore:
```diff
@@ -80,7 +80,6 @@ typings/
 
 # Nuxt.js build / generate output
 .nuxt
-dist
```
Just curious - do you mean to have the `dist/` directory in the repo?

sbrudz replied:
For typescript actions, it turns out that the compiled javascript needs to be committed to the repo ☹️ .  I'm using [ncc](https://github.com/vercel/ncc) to compile everything to an index.js file in the dist folder. (following the pattern set by the [ts action template](https://github.com/actions/typescript-action) from github)

sbrudz committed:
fix: skip deploys with commits that don't exist when calculating lead time

sbrudz committed:
fix: regenerate compiled dist files

so that changes are available to consumers of action

sbrudz committed:
fix: import ejs as a string instead of an ejs file

so that it will get included in dist/index.js file, preventing errors in action callers

sbrudz committed:
ci: use node v12 when running CI build for action

sbrudz committed:
docs: update readme with github action information

sbrudz committed:
refactor: remove git_project_directory parameter

because it is not needed

sbrudz committed:
refactor: delete report.ejs because it has been replaced by report-template.ts

sbrudz committed:
chore: remove dotenv-related files because they're no longer used

sbrudz committed:
chore: bump version to 1.0 to prepare for release

sbrudz committed:
docs: add thank you to Rhino to ReadMe

sbrudz committed:
docs: add some newly discovered similar projects
